File: projects/agent_based_modelling/assignment_4/model/road/car/car.go
package car

type Car struct {
	Velocity, MaxVelocity int
}

func New(maxVelocity int) *Car {
	return &Car{MaxVelocity: maxVelocity}
}

func (c *Car) Accelerate() {
	if c.Velocity < c.MaxVelocity {
		c.Velocity += 1
	}
}

func (c *Car) Decelerate() {
	if c.Velocity > 0 {
		c.Velocity -= 1
	}
}

File: projects/agent_based_modelling/assignment_4_cpp/Car.cpp
#include "Car.h"


Car::Car(const int max_velocity): max_velocity(max_velocity) {}

void Car::accelerate() {
    if (velocity < max_velocity) {
        velocity++;
    }
}

void Car::decelerate() {
    if (velocity > 0) {
        velocity = velocity - 1;
    }
}
